java/core/src/java/org/apache/orc/impl/writer/StringBaseTreeWriter.java

```diff
@@ -64,6 +64,15 @@
                        WriterEncryptionVariant encryption,
                        WriterContext writer) throws IOException {
     super(schema, encryption, writer);
+    Configuration conf = writer.getConfiguration();
+
```

Comment:
I'd make a function that looks at the configuration value and builds the hash or red-black dictionary directly. I can't see a need to make the dictionary mechanism pluggable.